### DISCUSSION TITLE
Docs: deprecate deploy tokens for Gitlab or clarify use in docs

### DISCUSSION THREAD

**Discussion by skycaptain** [renovatebot/renovate#30603] [Suggest an Idea]
### Tell us more.

Deploy tokens are limited to [this scope](https://docs.gitlab.com/ee/user/project/deploy_tokens/#scope). However, [the docs](https://docs.renovatebot.com/modules/platform/gitlab/#authentication) suggest that deploy tokens are a suitable way of defining `RENOVATE_TOKEN`. This is misleading because deploy tokens lack `read_api` and `api` permissions.

**Comment by viceice** [comment]
feel free to send a PR to fix it 🤗

**Reply by HonkingGoose** [reply]
Before creating a PR, let's get some more info! :wink: 

---

Hi @mbrodala :wave: 

You added the info about Deploy tokens to Renovate's GitLab Authentication docs with this PR:

- https://github.com/renovatebot/renovate/pull/30042

But a user says that the Deploy token is bad for `RENOVATE_TOKEN`, as it lacks the `read_api` and `api` permissions.

Can you tell us _how_ you're using the Deploy token? Are you maybe setting extra permissions in some config file? Or doing something else to get it working?

Thank you! :smile:

**Reply by skycaptain** [reply]
I think it's mentioned in the PR between the lines: *Registry authentication* works with a deploy token. Thus, a deploy token could work for a host rule but not as a general alternative for a PAT or GAT. For example, a deploy token cannot create a MR nor push to a repository on GitLab.

Maybe, I could have added more context on what I find misleading:

First, [the docs](https://docs.renovatebot.com/modules/platform/gitlab/#authentication) starts with this correct sentence. 

> You can authenticate Renovate to GitLab, with a Personal Access Token, or Group Access Token.

BTW. _Project Access Tokens_ are also an valid option and work similar to _Group Access Token_s, except for being limited to just one project.

But then continues:

> To start, create either:
> 
> - a [Personal Access Token](https://docs.gitlab.com/ee/user/profile/personal_access_tokens.html) for the bot account
> - or a [Group Access Token](https://docs.gitlab.com/ee/user/group/settings/group_access_tokens.html#bot-users-for-groups) for the bot account
> - or a [Deploy Token](https://docs.gitlab.com/ee/user/project/deploy_tokens/index.html) for the bot account

which is wrong as _Group Access Tokens_ and _Deploy tokens_ are independent of user accounts ([ref](https://docs.gitlab.com/ee/user/group/settings/group_access_tokens.html#:~:text=Group%20access%20tokens%20are%20similar,if%20the%20limit%20is%20set.), [ref](https://docs.gitlab.com/ee/user/project/deploy_tokens/)), and thus can not be created _for the bot account_, while _Personal Access Tokens_ can. However to be precise, creating a _Group Access Tokens_ or a _Project Access Tokens_ _creates_ a bot account on-the-fly in the background.

Then the docs continue with the part I find misleading as this only works for _Personal Access Token_, _Group Access Token_ and _Project Access Tokens_:

> The bot account must have at least the Developer role in order to [create issues and merge requests](https://docs.gitlab.com/ee/user/permissions.html#project-members-permissions). If you are using automerge, the bot account must have the appropriate ["Allowed to merge" permission on the protected branch](https://docs.gitlab.com/ee/user/project/protected_branches.html#require-everyone-to-submit-merge-requests-for-a-protected-branch) of your projects. If merging is restricted to Maintainers, the bot account must have the Maintainer role.

I propose to:
- remove mentioning Deploy tokens in the above paragraph and mention them in the paragraph that starts with *If you're using a private [GitLab container registry](https://docs.gitlab.com/ee/user/packages/container_registry/), you must:*
- mention _Project Access Tokens_ in the above paragraph
- rephrase the wording as _Group Access Tokens_ and _Deploy Tokens_ are not related to user accounts

I volunteer to create a PR.

**Reply by HonkingGoose** [reply]
Please open a PR so we can review in context. 🙂

**Reply by skycaptain** [reply]
https://github.com/renovatebot/renovate/pull/30787

While writing, I also came across [GitLab bot security](https://docs.renovatebot.com/gitlab-bot-security/#gitlab-bot-security), which has similar content that is largely outdated. I guess that requires another discussion/PR.

**Reply by HonkingGoose** [reply]
I'm thinking of getting permission from the author to get a better article into the docs:

- https://github.com/renovatebot/renovate/discussions/30446

Im waiting for the go-ahead from the Renovate maintainers to ask the author. 🙂

Do you want to read the article I'm thinking of pulling in? And tell me if the article is better than what we have now? 🙂
